spot-client/src/common/remote-control/BaseRemoteControlService.js

```diff
@@ -392,11 +460,40 @@ export class BaseRemoteControlService extends Emitter {
      * the message.
      * @param {string} from - The JID of the participant sending the message.
      * @param {Object} data - Additional information attached to the message.
+     * @protected
+     * @returns {void}
+     */
+    _processMessage(messageType, from, data) {
+        if (messageType === MESSAGES.REMOTE_CONTROL_P2P) {
+            this._processRemoteControlP2PMessage({
+                from,
+                data
+            });
+        }
+    }
+
+    /**
+     * Processing for {@link MESSAGES.REMOTE_CONTROL_P2P} message which are used to exchange offer/answer and ICE
+     * candidates needed to establish a P2P signaling channel.
+     *
+     * @param {string} from - The remote address from which the message has been received.
+     * @param {string} data - Any message data.
      * @private
      * @returns {void}
      */
-    _processMessage() {
-        return;
+    _processRemoteControlP2PMessage({ from, data }) {
+        if (!this._p2pSignaling) {
+            this._createP2PSignalingConnection(/* a server type of connection */ true);
+        }
+
+        if (this._p2pSignaling) {
+            this._p2pSignaling.processP2PMessage({
```

Comment:
What happens if the p2p signaling connection isn't ready after it was created a few lines above?

Reply:
It is not ready. Those messages here are to exchange offer/answer/ice candidates. The connection is ready only after ice connects. This part of the flow is really server specific where it will start the instance when first message arrives which is most likely an invite.

Reply:
This code is not only server specific. Both P2P client and server need to exchange either offer or answer and both process ice candidates.

Reply:
Okay so I misunderstood and the part above which creates the connection is server specific, but the part where a message is being processed is not.